src/core/Service.js

```diff
@@ -126,4 +126,37 @@ export default class Service extends EventEmitter {
 
     this.ros.callOnConnection(call);
   }
+
+  /**
+   * An alternate form of Service advertisement that supports a modern Promise-based interface for use with async/await.
+   * @param {(request: TRequest) => Promise<TResponse>} callback An asynchronous callback processing the request and returning a response.
+   */
+  advertiseAsync(callback) {
+    if (this.isAdvertised) {
+      throw new Error('Cannot advertise the same Service twice!');
+    }
```

Comment:
I think we should move toward this behavior in all of the cases throughout the library where there's currently guard conditions that just early-return with no information. Users shouldn't be calling `advertise` multiple times on the same `Service` object, that's probably an issue with their code!

I don't want to break that in existing functions yet, but figured I'd do it here where we have a green field.

Reply:
We have not released any v2 version yet. So I think we can fix it everywhere.